cmd/teleportercmd/relayercmd/deploy.go

```diff
@@ -0,0 +1,386 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package relayercmd
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/cobrautils"
+	"github.com/ava-labs/avalanche-cli/pkg/contract"
+	"github.com/ava-labs/avalanche-cli/pkg/evm"
+	"github.com/ava-labs/avalanche-cli/pkg/localnet"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/networkoptions"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/teleporter"
+	"github.com/ava-labs/avalanche-cli/pkg/utils"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanche-cli/pkg/vm"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/spf13/cobra"
+)
+
+type DeployFlags struct {
+	Network networkoptions.NetworkFlags
+	Version string
+}
+
+var (
+	deploySupportedNetworkOptions = []networkoptions.NetworkOption{
+		networkoptions.Local,
+		networkoptions.Devnet,
+		networkoptions.Fuji,
+	}
+	deployFlags DeployFlags
+)
+
+// avalanche teleporter relayer deploy
+func newDeployCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Deploys an ICM Relayer for the given Network",
+		Long:  `Deploys an ICM Relayer for the given Network.`,
+		RunE:  deploy,
+		Args:  cobrautils.ExactArgs(0),
+	}
+	networkoptions.AddNetworkFlagsToCmd(cmd, &deployFlags.Network, true, deploySupportedNetworkOptions)
+	cmd.Flags().StringVar(&deployFlags.Version, "version", "latest", "version to deploy")
+	// flag for binary to use
+	// flag for local process vs cloud
+	// flag for provided config file (may need to change tmp dir)
+	return cmd
+}
+
+func deploy(_ *cobra.Command, args []string) error {
+	return CallDeploy(args, deployFlags)
+}
+
+func CallDeploy(_ []string, flags DeployFlags) error {
+	network, err := networkoptions.GetNetworkFromCmdLineFlags(
+		app,
+		"In which Network will operate the Relayer?",
+		flags.Network,
+		true,
+		false,
+		deploySupportedNetworkOptions,
+		"",
+	)
+	if err != nil {
+		return err
+	}
+
+	deployToRemote := false
+	if network.Kind != models.Local {
```

Comment:
why do we want to deploy relayer to local for fuji clusters ?

Reply:
(1) to best support user cases of local fuji nodes (2) to easy relayer testing in general without requiring a remote node